### PR TITLE
feat: add daedalus-language-server support

### PR DESCRIPTION
**Pull request by pm4rcin**
This PR adds support for https://github.com/kirides/DaedalusLanguageServer used by Gothic and Gothic II games. Please let me know if the `root_dir` was added correctly as I'm not sure if it's equivalent of steps described in README of mentioned repo.

### CODE REVIEW

File: lua/lspconfig/server_configurations/daedalus_ls.lua [hunk 1]
```diff
@@ -0,0 +1,33 @@
+local util = require 'lspconfig.util'
+local bin_name = 'DaedalusLanguageServer'
+
+if vim.fn.has 'win32' == 1 then
+  bin_name = bin_name + '.exe'
+end
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/daedalus_ls.lua:6]:
```suggestion

```

**Review by glepnir — COMMENTED** [review]
you don't need do this now.

File: lua/lspconfig/server_configurations/daedalus_ls.lua [hunk 2]
```diff
@@ -0,0 +1,33 @@
+local util = require 'lspconfig.util'
+local bin_name = 'DaedalusLanguageServer'
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/daedalus_ls.lua:2]: inline

File: lua/lspconfig/server_configurations/daedalus_ls.lua (hunk 1 above)
Comment (justinmk) [reply lua/lspconfig/server_configurations/daedalus_ls.lua:6]: also, if this line were kept, the operator is wrong (should be `..`)

Comment (pm4rcin) [reply lua/lspconfig/server_configurations/daedalus_ls.lua:6]: May I ask why should I remove it? I've checked some servers configs in repo and they had that. Or is it automatic now?

File: lua/lspconfig/server_configurations/daedalus_ls.lua (hunk 2 above)
Comment (pm4rcin) [reply lua/lspconfig/server_configurations/daedalus_ls.lua:2]: I'm not sure what the request is but I guess I should add a newline between right? If yes then why you call it like that?

File: lua/lspconfig/server_configurations/daedalus_ls.lua (hunk 1 above)
Comment (glepnir) [reply lua/lspconfig/server_configurations/daedalus_ls.lua:6]: which servers ?

Comment (glepnir) [reply lua/lspconfig/server_configurations/daedalus_ls.lua:6]: oh I see. they are also should be removed i will do it later. we use exepath to find server executable so no need it now

File: lua/lspconfig/server_configurations/daedalus_ls.lua (hunk 2 above)
Comment (glepnir) [reply lua/lspconfig/server_configurations/daedalus_ls.lua:2]: I mean can just in cmd after win32 condition removed.

Comment (pm4rcin) [reply lua/lspconfig/server_configurations/daedalus_ls.lua]: Of course! I didn't see it since you highlighted those 2 lines together :D Fixed.

**Comment by glepnir** [comment]
thanks

File: lua/lspconfig/server_configurations/daedalus_ls.lua (hunk 1 above)
Comment (pm4rcin) [reply lua/lspconfig/server_configurations/daedalus_ls.lua]: Done.

**Pull request merged**